apps/lib/apps_ui.c

```diff
@@ -99,18 +99,31 @@ static int ui_close(UI *ui)
     return 1;
 }
 
+static char *(ui_prompt_construct)(UI *ui, const char *object_desc,
+                                      const char *object_name)
+{
+    PW_CB_DATA *cb_data = (PW_CB_DATA *)UI_get0_user_data(ui);
+
+    return cb_data != NULL
+        ? UI_construct_prompt(NULL, "pass phrase", cb_data->prompt_info)
+        : UI_construct_prompt(NULL, object_desc, object_name);
```

Comment:
That's not at all obvious to me.  How about this?

``` C
    PW_CB_DATA *cb_data = (PW_CB_DATA *)UI_get0_user_data(ui);

    if (object_desc == NULL)
        object_desc = "pass phrase";
    if (object_name == NULL)
        object_name = (cb_data == NULL ? cb_data->prompt_info : NULL);
    return UI_construct_prompt(NULL, object_desc, object_name);
```

Reply:
> That's not at all obvious to me. How about this?

I've taken over the essence of this, correcting `cb_data == NULL` to `cb_data != NULL`.

Reply:
On this occasion I've renamed the `object_desc` parameter to the more adequate `phrase_desc`,
also for the `UI_construct_prompt()` function, and improved its documentation.
For instance, so far it was implicit that  `object_name` may be NULL, in which case the `for <object_name>` is left out.

Reply:
Hmmm, `object_desc` is meant to be a description of the object that's going to be encrypted/decrypted, not a description of the phrase.

Reply:
Never mind that last comment, I wasn't thinking right